src/software/backend/simulation/simulator_robot.cpp

```diff
@@ -1,382 +1,437 @@
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
+    {
+        robot->brakeMotorBackRight();
+    }
+    else
     {
-        // TODO: Implement me
+        LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot"
+                     << std::endl;
     }
 }
 
-void SimulatorRobotSingleton::brakeMotorFrontRight()
+void SimulatorRobot::brakeMotorFrontRight()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
+    {
+        robot->brakeMotorFrontRight();
+    }
+    else
     {
-        // TODO: Implement me
+        LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot"
+                     << std::endl;
     }
 }
 
-std::weak_ptr<PhysicsRobot> SimulatorRobotSingleton::getCurrentPhysicsRobot()
+void SimulatorRobot::onChickerBallContact(PhysicsRobot *physics_robot,
+                                          PhysicsBall *physics_ball)
 {
-    if (!robot_id.has_value())
+    if (autokick_speed_m_per_s)
+    {
+        Vector kick_vector = Vector::createFromAngle(
+            physics_robot->getRobotWithTimestamp(Timestamp::fromSeconds(0))
+                .orientation());
+        kick_vector = kick_vector.normalize(autokick_speed_m_per_s.value());
+        physics_ball->kick(kick_vector);
+    }
+    else if (autochip_distance_m)
     {
-        return std::weak_ptr<PhysicsRobot>();
+        Vector chip_vector = Vector::createFromAngle(
+            physics_robot->getRobotWithTimestamp(Timestamp::fromSeconds(0))
+                .orientation());
+        chip_vector = chip_vector.normalize(autochip_distance_m.value());
+        physics_ball->chip(chip_vector);
     }
+}
 
-    unsigned int robot_id_value = *robot_id;
-    auto robot_id_comparator = [robot_id_value](std::weak_ptr<PhysicsRobot> robot_ptr) {
-        if (auto robot_ptr_lock = robot_ptr.lock())
-        {
-            return robot_ptr_lock->getRobotId() == robot_id_value;
-        }
-        return false;
-    };
-    auto physics_robot_iter =
-        std::find_if(physics_robots.begin(), physics_robots.end(), robot_id_comparator);
-    if (physics_robot_iter == physics_robots.end())
+void SimulatorRobot::onDribblerBallContact(PhysicsRobot *physics_robot,
+                                           PhysicsBall *physics_ball)
+{
+    if (dribbler_rpm > 0)
     {
-        return std::weak_ptr<PhysicsRobot>();
+        auto robot = physics_robot->getRobotWithTimestamp(Timestamp::fromSeconds(0));
+        auto ball  = physics_ball->getBallWithTimestamp(Timestamp::fromSeconds(0));
+
+        // To dribble, we apply a force towards the center and back of the dribbling area,
+        // closest to the chicker. We vary the magnitude of the foce by how far the ball
+        // is from this "dribbling point". This more-or-less acts like a tiny gravity well
+        // that sucks the ball into place, except with more force the further away the
+        // ball is
```

Comment:
We should make sure we mention that this goes away the moment the ball escapes the dribbler

Reply:
good idea. changed.